### ISSUE TITLE
Support publishing for many bun workspaces at same time

### ISSUE THREAD

**Issue by FranciscoCaetano88** [oven-sh/bun#5050]
### What is the problem this feature would solve?

Add a build system for managing and publishing multiple JavaScript/TypeScript packages from the same repository.

### What is the feature you are proposing to solve the problem?

A package versioning & publishing manager that should provide a variety of options to make sure any workflow can be accommodated.

### What alternatives have you considered?

Lerna

**Comment by The-Code-Monkey** [comment]
Yeah this is something I am missing from Bun as i currently use an old version of lerna to publish packages from my Monorepo would be nice to see something similar baked into bun +1

**Comment by nyxb** [comment]
Yes please I would also like a bun publish command so that I don't have to use npm, pnpm or yarn to publish packages. So I am forced to continue to have node installed in parallel

**Comment by vxhly** [comment]
Yes please I would also like a bun publish command.

The pnpm command I am using now seems not fully compatible with bun.

Below is the pnpm release process I listed, I hope bun will take this process into consideration.

```json
{
  "scripts": {
    "commit:changelog": "git add CHANGELOG.md && git commit -m \"docs: :memo: update changelog\" && git push",
    "lint": "eslint --ext mjs,js,jsx,ts,tsx,cjs --fix .",
    "release": "pnpm release:check && pnpm release:version && pnpm release:publish",
    "release:changelog": "conventional-changelog -p angular -i CHANGELOG.md -s && npm run commit:changelog",
    "release:check": "pnpm lint",
    "release:publish": "pnpm -r publish --tag latest",
    "release:version": "bumpp package.json packages/*/package.json --execute=\"pnpm release:changelog\" --commit \"build: :package: publish v%s\" --all"
  },
  "devDependencies": {
    "bumpp": "^9.2.0",
    "conventional-changelog-cli": "^4.1.0",
  }
}
```

**Comment by vxhly** [comment]
`bun` seems to be missing functions like `pnpm update -r --workspace`, and I hope it can be added in the next version.

**Comment by oagix** [comment]
Has anybody been able to figure out a way to publish a package using bun?

**Comment by The-Code-Monkey** [comment]
@oalexdoda still using lerna to publish for now.

@Jarred-Sumner could you weigh in on this, as this feature isn't listed in the roadmap?

**Comment by The-Code-Monkey** [comment]
Any update on this?

**Comment by salmenf** [comment]
Bun `publish`/`deprecate`/`unpublish` would be great! I was surprised this wasn't part of Bun already, since you can in fact provide login credentials for registries in your bunfig.

The only workaround I know of is to run an external script handling publishing (e.g. https://github.com/morlay/bunpublish). This isn't great since it introduces extra dependencies.

**Comment by pipinet** [comment]
Any update on this?

**Comment by StephenHaney** [comment]
It looks like `bun publish` is [now a thing](https://github.com/oven-sh/bun/blob/main/docs/cli/publish.md).

I'm confused by how it maps workspace dependencies (like `workspace:*`) though – it seems to always lag 2 versions behind what it should use. Still testing to try to narrow down what is happening.

**Comment by janwirth** [comment]
Ping.

I want to develop multiple packages that work best together with bun, but I'm not sure how to best keep them in the same place, while they depend on each other. Will the links between the workspaces automatically be updated when I use `publish`? This isn't clear.

**Comment by janwirth** [comment]
Actually just found this:

<img width="791" alt="Image" src="https://github.com/user-attachments/assets/7b070219-bb56-41cb-95b2-c051f6f4bd68" />

**Comment by mimikn** [comment]
I stumbled across this ticket while searching for possible solutions to the problem.

The issue right now when I want to publish multiple packages in a monorepo that are linked together is that : `bun publish` will do a `bun pm pack` that will check all dependencies with the `workspace:` protocol and try to replace it with what is in the lockfile BUT after I bumped my packages versions, there's no way to update the lockfile with the new versions of the packages.

The only way I found is to run a `bun update [your workspace deps]` in all packages that contain workspace dependencies to force the lockfile to be updated with the new versions.

I think it would be a good idea to update the lockfile with a command like `bun install` if workspaces packages versions have changed, or at least to have a command that would update the lockfile with the new versions of the packages.

**Comment by pipinet** [comment]
Any update on this?
